frontend/src/containers/ChooseStaticDataset.tsx

```diff
@@ -0,0 +1,145 @@
+import React, { useState, useCallback } from "react";
+import Breadcrumbs from "../components/Breadcrumbs";
+import { useHistory, useParams } from "react-router-dom";
+import {
+  useDashboard,
+  useDateTimeFormatter,
+  useSettings,
+  useDatasets,
+} from "../hooks";
+import Table from "../components/Table";
+import Button from "../components/Button";
+import Search from "../components/Search";
+import ScrollTop from "../components/ScrollTop";
+import { Dataset } from "../models";
+
+interface PathParams {
+  dashboardId: string;
+}
+
+function ChooseStaticDataset() {
+  const history = useHistory();
+  const dateFormatter = useDateTimeFormatter();
+
+  const { dashboardId } = useParams<PathParams>();
+  const { dashboard } = useDashboard(dashboardId);
+  const { staticDatasets } = useDatasets();
+  const { settings } = useSettings();
+
+  const [filter, setFilter] = useState("");
+  const [selected, setSelected] = useState<Array<Dataset>>([]);
+
+  const crumbs = [
+    {
+      label: "Dashboards",
+      url: "/admin/dashboards",
+    },
+    {
+      label: dashboard?.name,
+      url: `/admin/dashboard/${dashboardId}`,
+    },
+    {
+      label: "Add chart",
+      url: `/admin/dashboard/${dashboardId}/add-chart`,
+    },
+    {
+      label: "Choose static dataset",
+      url: "",
+    },
+  ];
+
+  const onSearch = (query: string) => {
+    setFilter(query);
+  };
+
+  const onCancel = () => {
+    history.goBack();
+  };
+
+  const onSelect = useCallback((selectedDataset: Array<Dataset>) => {
+    setSelected(selectedDataset);
+  }, []);
+
+  const onSubmit = () => {
+    history.replace({
+      pathname: `/admin/dashboard/${dashboardId}/add-chart`,
```

Comment:
How are we going to reuse this screen for other types of content items? I think this screen needs to be agnostic of what type of content item the user is adding. Should be easy to do though, just have this screen receive the `redirectUrl` as part of the state, and then when the user selects a dataset, you redirect them back to where they came from.

Reply:
That is a very good point. Let me fix that.